lib/rules/no-compare-neg-zero.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * @fileoverview The rule should warn against code that tries to compare against -0.
+ * @author no-compare-neg-zero
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "disallow compare against -0.",
+            category: "Possible Errors",
+            recommended: false
+        },
+        fixable: null,
+        schema: []
+    },
+
+    create(context) {
+
+        //--------------------------------------------------------------------------
+        // Helpers
+        //--------------------------------------------------------------------------
+
+        /**
+         * Checks a given node is -0
+         *
+         * @param {ASTNode} node - A node to check.
+         * @returns {boolean} `true` if the node is -0.
+         */
+        function isNegZero(node) {
+            return node.type === "UnaryExpression" && node.operator === "-" && node.argument.type === "Literal" && node.argument.value === 0;
+        }
+
+        return {
+            BinaryExpression(node) {
+                if (/^(?:[<>]|[!=]=)=?$/.test(node.operator)) {
```

Comment:
This is a small nitpick that shouldn't block this PR from being merged, but I think this regex is sort of hard to read. Since we're looking for a static list of operators, it might be more readable to do something like

```js
const OPERATORS_TO_CHECK = new Set([">", ">=", "<", "<=", "==", "===", "!=", "!=="]);

return BinaryExpression(node) {
  if (OPERATORS_TO_CHECK.has(node.operator)) {
    // ...
  }
}
```